test/tests/CMakeLists.txt

```diff
@@ -266,3 +266,55 @@ SET_CHECK_CXX_FLAGS(test_s6importexporttests)
 target_link_libraries(test_s6importexporttests ${GTEST_LIBRARIES} libopenrct2 ${LDL} z)
 target_link_platform_libraries(test_s6importexporttests)
 add_test(NAME s6importexporttests COMMAND test_s6importexporttests)
+
+# Vehicle test
+set(VEHICLE_TEST_SOURCES
+        "${CMAKE_CURRENT_LIST_DIR}/VehicleTest.cpp"
+        "${ROOT_DIR}/src/openrct2/ride/Vehicle.cpp"
+        "${ROOT_DIR}/src/openrct2/ride/VehicleSubpositionData.cpp"
+        "${ROOT_DIR}/src/openrct2/ride/VehicleData.cpp"
+        "${ROOT_DIR}/src/openrct2/world/Sprite.cpp"
+        "${ROOT_DIR}/src/openrct2/world/TileElementBase.cpp"
+        "${ROOT_DIR}/src/openrct2/ride/CableLift.cpp"
+        "${ROOT_DIR}/src/openrct2/ride/RideData.cpp"
+        "${ROOT_DIR}/src/openrct2/ride/TrackData.cpp"
+        "${ROOT_DIR}/src/openrct2/ride/Track.cpp"
+        "${ROOT_DIR}/src/openrct2/ride/Station.cpp"
+        "${ROOT_DIR}/src/openrct2/windows/Intent.cpp"
+        "${ROOT_DIR}/src/openrct2/core/Crypt.OpenSSL.cpp"
+        #"${ROOT_DIR}/src/openrct2/audio/Audio.cpp"
```

Comment:
These were temporaries I tried using initially, to figure out how far can I reduce the necessary data